src/sql/workbench/contrib/notebook/browser/outputs/gridOutput.component.ts

```diff
@@ -119,6 +119,29 @@ export class GridOutputComponent extends AngularDisposable implements IMimeCompo
 		}
 		if (!this._table) {
 			let source = <IDataResource><any>this._bundleOptions.data[this.mimeType];
+			let columnNames: Array<string> = [];
+			let rowIndex = 0;
+			// Get schema list of grid
+			for (let i of source.schema.fields) {
+				columnNames.push(i.name);
+			}
+			// Check to see if data source is ordered properly based on schema
+			// SQL notebooks do not use columnName as key (instead uses indices)
+			// So we add a condition for SQL notebooks to not be reordered
+			if (source.data.length > 0) {
+				if (columnNames !== Object.keys(source.data[0]) && Object.keys(source.data[0])[0] !== '0') {
+					// Order each row based on the schema
+					for (let row of source.data) {
+						let reorderedData = {};
+						for (let key of columnNames) {
```

Comment:
So I'm curious how this handles columns with the same name - column names aren't guaranteed to be unique

Reply:
Dictionaries of the rows should enforce this, the object of the row is in form of a dictionary as such keys in dictionaries (which would be the columnNames) should always be unique.

Reply:
So what happens when you run a query with multiple columns that have the same name?

`SELECT 1 as col, 2 as col`

Reply:
Since this fix is targeted only for KQL rendered grids this would not be possible to do in Kusto. You are not allowed to have the same column names in the same table in Kusto, it would result in a query execution error.

Reply:
How is this only affecting KQL grids? Can you point me to where this is being only used for that vs all other result grids?

Reply:
That is what the condition is there for. It's directly related to your comment above for the check against '0' which was my naive method to check for SQL notebooks and not reorder them. I will have a better condition however for SQL notebooks as that was not robust.

Reply:
Let's put that in a comment so it's clear that this precondition needs to be true for this logic to work (i.e. we can't just apply this to SQL tables )

How do SQL tables even come in as? They can't be an object like this because object properties can't have the same name so there must be some other format they come in as right?